### PR TITLE
fix docstrings for `data_iterator` and `data_source`

### PR DESCRIPTION
**Pull request by faroit**
The documentation/doc strings mention `data_source_design` and `data_iterator_design` which don't exist. I guess the following fix is what was intended instead.

### CODE REVIEW

File: python/src/nnabla/utils/data_iterator.py [hunk 1]
```diff
@@ -38,9 +38,9 @@
 
 class DataIterator(object):
     '''DataIterator
-    Collect data from :ref:`data_source_design` and yields bunch of data.
+    Collect data from :ref:`data_source` and yields bunch of data.
```
[the remark this answers is not in the document]
Comment (faroit) [reply python/src/nnabla/utils/data_iterator.py]: done

File: python/src/nnabla/utils/data_iterator.py [hunk 2]
```diff
@@ -38,9 +38,9 @@
...
 
-    Detailed documentation is available in :ref:`data_iterator_design`.
+    Detailed documentation is available in :ref:`data_iterator`.
```
[the remark this answers is not in the document]
Comment (faroit) [reply python/src/nnabla/utils/data_iterator.py]: done

**Comment by TE-TakuyaYashima** [comment]
Thanks for your cooperation!

**Comment by faroit** [comment]
You are welcome

**Pull request merged**